unified-runtime/source/adapters/level_zero/common.hpp

```diff
@@ -435,6 +504,9 @@ struct _ur_object {
   // Indicates if we own the native handle or it came from interop that
   // asked to not transfer the ownership to SYCL RT.
   bool OwnNativeHandle = false;
+
+  // Indicates if this object is an interop handle.
+  bool IsInteropNativeHandle = false;
```

Comment:
Do we need to add a new variable? Can we just adjust OwnNativeHandle in the constructors?

Reply:
So, we have to have a second variable because we have overloaded the variable to say both:
1. we own this and created this and can delete it
2. we have been given ownership of this and can delete it.

"// Indicates if we own the native handle or it came from interop"

So, if we created this handle, then we need to call destroy, otherwise we will appear to have a memory leak with ourselves.

However, if we did not create this handle, then we "can" try to destroy it only if the loader is stable.

Reworking the overloaded variable would be a larger code change and could inadvertently cause problems in then logic so I opted to add another to specifically check for the one case.

Reply:
Right, but for the first case, OwnNativeHandle would always be false. But in any case, if you prefer to have a separate variable I'm ok with that - especially since this is hidden in the raii handle for v2.